framework/src/modules/chain/forger.js

```diff
@@ -183,89 +181,76 @@ class Forger {
-						parseEncryptedPassphrase(encryptedItem.encryptedPassphrase),
-						this.config.forging.defaultPassword
-					);
-				} catch (error) {
-					return setImmediate(
-						seriesCb,
-						`Invalid encryptedPassphrase for publicKey: ${
-							encryptedItem.publicKey
-						}. ${error.message}`
-					);
-				}
-
-				const {
-					publicKeyBytes,
-					privateKeyBytes,
-				} = getPrivateAndPublicKeyBytesFromPassphrase(passphrase);
-
-				const keypair = {
-					publicKey: publicKeyBytes,
-					privateKey: privateKeyBytes,
-				};
-
-				if (keypair.publicKey.toString('hex') !== encryptedItem.publicKey) {
-					return setImmediate(
-						seriesCb,
-						`Invalid encryptedPassphrase for publicKey: ${
-							encryptedItem.publicKey
-						}. Public keys do not match`
-					);
-				}
-
-				const filters = {
-					address: getAddressFromPublicKey(keypair.publicKey.toString('hex')),
-				};
-
-				const options = {
-					extended: true,
-				};
-
-				return this.storage.entities.Account.get(filters, options)
-					.then(accounts => {
-						const account = accounts[0];
-						if (!account) {
-							return setImmediate(
-								seriesCb,
-								`Account with public key: ${keypair.publicKey.toString(
-									'hex'
-								)} not found`
-							);
-						}
-						if (account.isDelegate) {
-							this.keypairs[keypair.publicKey.toString('hex')] = keypair;
-							this.logger.info(
-								`Forging enabled on account: ${account.address}`
-							);
-						} else {
-							this.logger.warn(
-								`Account with public key: ${keypair.publicKey.toString(
-									'hex'
-								)} is not a delegate`
-							);
-						}
-
-						return setImmediate(seriesCb);
-					})
-					.catch(err => setImmediate(seriesCb, err));
-			},
-			cb
-		);
+		// eslint-disable-next-line no-restricted-syntax
+		for (const encryptedItem of encryptedList) {
+			let passphrase;
+			try {
+				passphrase = decryptPassphraseWithPassword(
+					parseEncryptedPassphrase(encryptedItem.encryptedPassphrase),
+					this.config.forging.defaultPassword
+				);
+			} catch (error) {
+				const decryptionError = `Invalid encryptedPassphrase for publicKey: ${
+					encryptedItem.publicKey
+				}. ${error.message}`;
+				this.logger.error(decryptionError);
+				throw decryptionError;
+			}
+
+			const {
+				publicKeyBytes,
+				privateKeyBytes,
+			} = getPrivateAndPublicKeyBytesFromPassphrase(passphrase);
+
+			const keypair = {
+				publicKey: publicKeyBytes,
+				privateKey: privateKeyBytes,
+			};
+
+			if (keypair.publicKey.toString('hex') !== encryptedItem.publicKey) {
+				throw `Invalid encryptedPassphrase for publicKey: ${
+					encryptedItem.publicKey
+				}. Public keys do not match`;
+			}
+
+			const filters = {
+				address: getAddressFromPublicKey(keypair.publicKey.toString('hex')),
+			};
+
+			const options = {
+				extended: true,
+			};
+
+			// eslint-disable-next-line no-await-in-loop
+			const accounts = await this.storage.entities.Account.get(
```

Comment:
```suggestion
			const [account] = await this.storage.entities.Account.get(
```

Reply:
Committed and pushed locally as I wanted to run tests.